homeassistant/components/shelly/config_flow.py

```diff
@@ -463,11 +459,7 @@ def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlowHandler:
     @callback
     def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         """Return options flow support for this handler."""
-        return (
-            get_device_entry_gen(config_entry) in RPC_GENERATIONS
-            and not config_entry.data.get(CONF_SLEEP_PERIOD)
-            and config_entry.data.get("model") != MODEL_WALL_DISPLAY
-        )
+        return cast(bool, config_entry.data.get(CONF_SCRIPT, False))
```

Comment:
Options flow is not available after configuring a device that supports scripting.

![obraz](https://github.com/user-attachments/assets/73e57065-241c-411c-8e5a-0edeac8b2093)

Reply:
Humm, don't know what to say; works here:

![image](https://github.com/user-attachments/assets/3e87321d-b362-4bfe-9b3c-beed5d1db0b8)


Did you add the device after starting HA with this PR or did you start with the device already configured ?

Reply:
Found it, the issue is evident if you add the device after starting HA. Needs a fix.

BTW, if you restart Ha then all is fine for the previously added device.

Reply:
I think the only way to get it right on all situation is a migration of the `config_entry` to add the new info about CONF_SCRIPT